projects/netgrif-components-core/src/lib/task-content/task-content/abstract-task-content.component.ts

```diff
@@ -377,7 +405,36 @@ export abstract class AbstractTaskContentComponent implements OnDestroy {
             dataGroups.splice(firstNonDescendantIndex, 0, split.endGroup);
         }
 
-        return result;
+        return {
+            dataGroups: result,
+            containsDashboardTaskRef: containsDashboard
+        };
+    }
+
+    /**
+     * Creates a duplicate of the provided data group array.
+     *
+     * Only the data groups are cloned, the fields are only copied as references.
+     * @param dataGroups the data groups that should be cloned
+     * @returns the duplicated data groups
+     */
+    protected cloneDataGroups(dataGroups: Array<DataGroup>): Array<DataGroup> {
+        return dataGroups.map(g => g);
```

Comment:
this should clone the CONTENT of the data groups (i.e. the field arrays), so that they can be modified in a destructive way. This implementation passes the original field arrays trough and they are modified by the layout algorithms!